packages/nocodb/src/lib/db/sql-data-mapper/lib/sql/sortV2.ts

```diff
@@ -232,8 +232,19 @@ export default async function sortV2(
           break;
         }
       default:
-        qb.orderBy(sanitize(column.column_name), sort.direction || 'asc');
+      {
+        const clientType = knex.clientType();
+        if (clientType === 'mssql') {
+          qb
+            .orderBy(sanitize(knex.raw('LEN(??)', [column.column_name])))
+            .orderBy(sanitize(column.column_name), sort.direction || 'asc');
+        } else {
+          qb
+            .orderBy(sanitize(knex.raw('LENGTH(??)', [column.column_name])))
```

Comment:
Using this kind of sort we can generate inconsistencies with data.

![image](https://user-images.githubusercontent.com/11861080/193765953-2f9639ff-ac26-49d1-9c3d-fd196024c11a.png)

The issue touches all `varchar` datatypes. (the sort is **asc**, but considering the length we got this)

![image](https://user-images.githubusercontent.com/11861080/193766288-1bce0073-69bf-458b-8a7c-d3cd309b9478.png)